### ISSUE TITLE
Missing package dependencies

### ISSUE THREAD

**Issue by maciejskorski** [smathot/python-eyelinkparser#4]
The `pip` package as well as the source on git are missing dependencies, so that the package is not runnable with extra efforts. It would be nice to close this gap:-)
```
(.eyelinkparser) ubuntu@ip-172-31-36-77:~/projects/eyelinkparser/python-eyelinkparser$ pip install python-eyelinkparser
Collecting python-eyelinkparser
  Using cached python_eyelinkparser-0.17.1-py2.py3-none-any.whl (20 kB)
Collecting python-datamatrix
  Using cached python_datamatrix-0.14.3-py2.py3-none-any.whl (74 kB)
Collecting fastnumbers
  Using cached fastnumbers-3.2.1-cp310-cp310-manylinux_2_5_x86_64.manylinux1_x86_64.manylinux_2_12_x86_64.manylinux2010_x86_64.whl (87 kB)
Installing collected packages: python-datamatrix, fastnumbers, python-eyelinkparser
Successfully installed fastnumbers-3.2.1 python-datamatrix-0.14.3 python-eyelinkparser-0.17.1
(.eyelinkparser) ubuntu@ip-172-31-36-77:~/projects/eyelinkparser/python-eyelinkparser$ pip list
Package              Version
-------------------- -------
fastnumbers          3.2.1
pip                  22.0.2
python-datamatrix    0.14.3
python-eyelinkparser 0.17.1
setuptools           59.6.0
(.eyelinkparser) ubuntu@ip-172-31-36-77:~/projects/eyelinkparser/python-eyelinkparser$
```

**Comment by smathot** [comment]
Which dependencies do you mean exactly? The two dependencies that are specified, and which are also automatically installed in your example, are datamatrix and fastnumbers. Which additional dependencies are you missing?

**Comment by maciejskorski** [comment]
Child dependencies are missing, as shown above. 

So from a package we expect to install child dependencies (numpy, scipy, pretty etc) and run in a fresh environment without extra effort. Something like I did in my fork.

**Comment by smathot** [comment]
You're right actually. I had forgotten that numpy and scipy are only optional dependencies of datamatrix, but required for eyelinkparser.

**Comment by maciejskorski** [comment]
@smathot  the issue is not even about forgetting, but about enabling the CI/CD testing which would prevent proactively many things of that sort.
I created the fork for experimental purposes and it tests install & running in a fresh environment on a variety of Python versions. I am happy to contribute and help in that if you don't mind
https://github.com/maciejskorski/python-eyelinkparser

**Comment by smathot** [comment]
It looks like you've done a lot of useful work on the package! In principle, I'd be more than happy to merge some or all of these changes. However, would it be possible to split this up into separate PRs with a clear description so that I can review what exactly it does?

**Comment by Rockyneuron** [comment]
Hello!! I have arrived here through your paper 
https://doi.org/10.3758/s13428-022-01957-7

However I am not able to run the example from the documentation of the repository.

![image](https://github.com/open-cogsci/eyelinkparser/assets/82963868/6ed49796-1f26-4140-97d8-cafb12942c75)

Is there a .yml file or someting to use a similar environment?
Thanks in advance

**Comment by smathot** [comment]
Hi @Rockyneuron ,

This question is not related to this issue (or to a bug more generally), and is best posted on the [support forum](https://forum.cogsci.nl/). Would you mind moving the conversation there?

— Sebastiaan
